test/extended/node/dra.go

```diff
@@ -0,0 +1,77 @@
+package node
+
+import (
+	"context"
+
+	g "github.com/onsi/ginkgo/v2"
+	o "github.com/onsi/gomega"
+
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/kubernetes/test/e2e/framework"
+	admissionapi "k8s.io/pod-security-admission/api"
+
+	exutil "github.com/openshift/origin/test/extended/util"
+)
+
+var _ = g.Describe("[sig-node][DRA][OCPFeatureGate:DynamicResourceAllocation]", func() {
+	defer g.GinkgoRecover()
+
+	oc := exutil.NewCLIWithPodSecurityLevel("dra-scheduling", admissionapi.LevelPrivileged)
+
+	g.Context("Dynamic Resource Allocation", func() {
+
+		g.It("should verify beta and alpha DRA APIs are disabled [apigroup:resource.k8s.io]", func(ctx context.Context) {
+			g.By("discovering available API versions for resource.k8s.io group")
+			discoveryClient := oc.AdminKubeClient().Discovery()
+			apiGroup, err := discoveryClient.ServerResourcesForGroupVersion("resource.k8s.io/v1")
+			o.Expect(err).NotTo(o.HaveOccurred(), "v1 API should be available when DRA feature gate is enabled")
+			o.Expect(apiGroup).NotTo(o.BeNil())
+			framework.Logf("Found resource.k8s.io/v1 API with %d resources", len(apiGroup.APIResources))
+
+			g.By("verifying v1alpha2 API is not available")
+			_, err = discoveryClient.ServerResourcesForGroupVersion("resource.k8s.io/v1alpha2")
+			o.Expect(err).To(o.HaveOccurred(), "v1alpha2 API should not be available")
+			o.Expect(apierrors.IsNotFound(err)).To(o.BeTrue(), "v1alpha2 should return NotFound error")
+			framework.Logf("Confirmed v1alpha2 API is disabled: %v", err)
+
+			g.By("verifying v1beta1 API is not available")
+			_, err = discoveryClient.ServerResourcesForGroupVersion("resource.k8s.io/v1beta1")
+			o.Expect(err).To(o.HaveOccurred(), "v1beta1 API should not be available")
+			o.Expect(apierrors.IsNotFound(err)).To(o.BeTrue(), "v1beta1 should return NotFound error")
+			framework.Logf("Confirmed v1beta1 API is disabled: %v", err)
+
+			g.By("verifying v1alpha1 API is not available")
+			_, err = discoveryClient.ServerResourcesForGroupVersion("resource.k8s.io/v1alpha1")
+			o.Expect(err).To(o.HaveOccurred(), "v1alpha1 API should not be available")
+			o.Expect(apierrors.IsNotFound(err)).To(o.BeTrue(), "v1alpha1 should return NotFound error")
+			framework.Logf("Confirmed v1alpha1 API is disabled: %v", err)
+
```

Comment:
do we need the above? what we are doing below is asserting that there is only one api version being served